bson/bsoncodec/default_value_decoders.go

```diff
@@ -740,6 +740,9 @@ func (dvd DefaultValueDecoders) objectIDDecodeType(dc DecodeContext, vr bsonrw.V
 		if err != nil {
 			return emptyValue, err
 		}
+		if oid, err = primitive.ObjectIDFromHex(str); err == nil {
```

Comment:
Can you add a comment above this line to clarify the behavior:
```
Accept both 12-byte string and pretty-printed 24-byte hex string formats.
```

Reply:
Sure, no problem

Reply:
Thanks! Please re-request a review from me once you've added this.

Reply:
done